core/src/apps/cardano/helpers/chunks.py

```diff
@@ -0,0 +1,83 @@
+from micropython import const
+from typing import TYPE_CHECKING
+
+from trezor import messages, protobuf
+from trezor.wire import ProcessError
+from trezor.wire.context import call as ctx_call
+
+if TYPE_CHECKING:
+    from typing import Generic, TypeVar, Union
+
+    Chunk = Union[
+        messages.CardanoTxInlineDatumChunk,
+        messages.CardanoTxReferenceScriptChunk,
+        messages.CardanoMessagePayloadChunk,
+    ]
+
+    C = TypeVar("C", bound=Chunk)
+else:
+    # typechecker cheat
+    Generic = (object,)
+    C = Chunk = 0
+
+MAX_CHUNK_SIZE = const(1024)
+
+
+def _get_chunks_count(size: int) -> int:
+    """Integer-only version of `ceil(size / MAX_CHUNK_SIZE)`."""
+    assert size >= 0
+    return 0 if size == 0 else (size - 1) // MAX_CHUNK_SIZE + 1
+
+
+def _validate_chunk(
+    chunk: Chunk,
+    chunk_index: int,
+    total_size: int,
+) -> None:
+    chunks_count = _get_chunks_count(total_size)
+    assert chunk_index < chunks_count
+
+    chunk_name = chunk.__class__.__name__
```

Comment:
This ends up printing just `"Invalid last Msg: Size inconsistent with total bytes",`, I guess i should just use "chunk" instead of the class name?

Reply:
I think using only `chunk` is okay.